bibtex-actions-latex.el

```diff
@@ -0,0 +1,118 @@
+;;; bibtex-actions-latex.el --- Latex adapter for bibtex-actions -*- lexical-binding: t; -*-
+;;
+;; Copyright (C) 2021 Bruce D'Arcus
+;;
+;; Author: Bruce D'Arcus <https://github.com/bdarcus>
+;; Maintainer: Bruce D'Arcus <https://github.com/bdarcus>
+;; Created: February 27, 2021
+;; License: GPL-3.0-or-later
+;; Version: 0.4
+;; Homepage: https://github.com/bdarcus/bibtex-actions
+;; Package-Requires: ((emacs "26.3"))
+;;
+;; This file is not part of GNU Emacs.
+;;
+;; This program is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+;;
+;;; Commentary:
+;;
+;; A small package that provides the functions required to use bibtex-actions
+;; with latex.
+;;
+;; Simply loading this file will enable manipulating the citations with
+;; commands provided by bibtex-actions.
+;;
+;;; Code:
+
+(require 'bibtex-actions)
+(require 'tex)
+(require 'reftex)
+
+(defcustom bibtex-actions-latex-cite-commands
+  '((("cite" "Cite" "parencite"
```

Comment:
I don't really use LaTeX citations directly anymore, so take my questions/feedback with suitable grain-of-salt ...

I know this is how this is often implemented; as a flat list of commands.

But in practice, not all commands are the same; that their relevance depends on which system one is using; for example, natbib vs biblatex.

If you're using one, do you really want to have access to the long list of options compared to the other?

Also, I don't know if there's any value in cross-fertilizing ideas, because org-cite is more general than latex, but here's what the current org-cite style selection UI looks like by default, using Emacs 28 group-function.

![image](https://user-images.githubusercontent.com/1134/134823460-b45ef8b8-b47e-4bf7-ac01-0af06ece8965.png)

Reply:
> I don't really use LaTeX citations directly anymore, so take my questions/feedback with suitable grain-of-salt ...
> 
> I know this is how this is often implemented; as a flat list of commands.
> 
> But in practice, not all commands are the same; that their relevance depends on which system one is using; for example, natbib vs biblatex.
> 
> If you're using one, do you really want to have access to the long list of options compared to the other?
> 
> Also, I don't know if there's any value in cross-fertilizing ideas, because org-cite is more general than latex, but here's what the current org-cite style selection UI looks like by default, using Emacs 28 group-function.
> 
> ![image](https://user-images.githubusercontent.com/1134/134823460-b45ef8b8-b47e-4bf7-ac01-0af06ece8965.png)

I don't like the flat list of commands either but since I don't need an interface, I am not sure what it should be. Maybe it is worth asking people who use the multiple command what would they like the interface to be on doom discord (what would be good forum?). 

I am actually not familiar with biblatex or natbib either but if there is an easy way to cut down the number of options that have to presented depending on the file, that would be good. But as above some input has to be come from who use it.

Reply:
Could just not worry about it now, and iterate once we have a starting point?

Reply:
> Could just not worry about it now, and iterate once we have a starting point?

Sounds good though responses can take time so asking earlyish would be better.

This is going to take at least a couple of weekends though to get into some decent shape.